dandischema/metadata.py

```diff
@@ -265,45 +265,116 @@ def validate(
+    :return: The Dandiset metadata instance migrated to the target DANDI schema version
+    :raises ValueError: If the provided instance doesn't have a `schemaVersion` field
+        that specifies a valid DANDI schema version
+    :raises ValueError: If `to_version` is not a valid DANDI schema version or an
+        allowed target schema
+    :raises ValueError: If the target DANDI schema version is lower than the DANDI
+        schema version of the provided instance
+    """
+
+    # ATM, we only support the latest schema version as a target. See definition of
+    # `ALLOWED_TARGET_SCHEMAS` for details
     if len(ALLOWED_TARGET_SCHEMAS) > 1:
         raise NotImplementedError(
             "ATM code below supports migration to current version only"
         )
+
+    # --------------------------------------------------------------
+    # Validate DANDI schema version provided in the metadata instance
+    # --------------------------------------------------------------
+    # DANDI schema version of the provided instance
+    instance_ver = obj.get("schemaVersion")
+    if instance_ver is None:
+        msg = (
+            "The provided Dandiset metadata instance, `obj`, does not have a "
+            "`schemaVersion` field, for specifying DANDI schema version."
+        )
+        raise ValueError(msg)
+    if not isinstance(instance_ver, str):
+        msg = (
+            "The provided Dandiset metadata instance, `obj`, has a non-string "
+            "`schemaVersion` field, for specifying DANDI schema version."
+        )
+        raise ValueError(msg)
+    # Check if `instance_ver` is a valid DANDI schema version
+    try:
+        # DANDI schema version of the provided instance in tuple form
+        instance_ver_tuple = version2tuple(instance_ver)
+    except ValueError as e:
+        msg = (
+            "The provided Dandiset metadata instance, `obj`, has an invalid "
+            "`schemaVersion` field, for specifying DANDI schema version."
+        )
+        raise ValueError(msg) from e
+    if instance_ver not in ALLOWED_INPUT_SCHEMAS:
+        raise ValueError(f"Current input schemas supported: {ALLOWED_INPUT_SCHEMAS}.")
+    # ----------------------------------------------------------------
+
+    # --------------------------------------------------------------
+    # Validate `to_version`
+    # --------------------------------------------------------------
+    # Check if `to_version` is a valid DANDI schema version
+    try:
+        # The target DANDI schema version in tuple form
+        target_ver_tuple = version2tuple(to_version)
+    except ValueError as e:
+        msg = (
+            "The provided target version, `to_version`, is not a valid DANDI schema "
+            "version."
+        )
+        raise ValueError(msg) from e
+
+    # Permit only allowed target schemas
     if to_version not in ALLOWED_TARGET_SCHEMAS:
         raise ValueError(f"Current target schemas: {ALLOWED_TARGET_SCHEMAS}.")
```

Comment:
```suggestion
        raise ValueError(f"Target version {to_version!r} not in supported target schemas: {ALLOWED_TARGET_SCHEMAS}")
```

Reply:
Thanks. Suggestion is taken in modified form.